plugins/3MFReader/ThreeMFWorkspaceReader.py

```diff
@@ -371,7 +371,8 @@ def preRead(self, file_name, show_dialog=True, *args, **kwargs):
         # Get quality type
         parser = ConfigParser(interpolation = None)
         parser.read_string(serialized)
-        quality_container_id = parser["containers"][str(_ContainerIndexes.Quality)]
+        index_map_version = _ContainerIndexes.getIndexMapping(int(parser["metadata"]["setting_version"]))
+        quality_container_id = parser["containers"][str(index_map_version[_ContainerIndexes.Quality])]
```

Comment:
I think here it's also missing the part that if this project file is from 4.3+, it will have intent containers, and here in the project file loading, it should use the intent containers specified in the project.

Reply:
I'll file this as a different bug. It looks like it's indeed not checking for collisions between the intents.